### ISSUE TITLE
[Feature] Overload waitForEvent to return specific type

### ISSUE THREAD

**Issue by kalekseev** [microsoft/playwright#1674]
Right now `waitForEvent` return type is `Object`, but we can have specific types with overload:

```
waitForEvent(event: "popup", optionsOrPredicate?: Function|PageWaitForEventOptionsOrPredicate): Promise<Page>;
...
waitForEvent(event: string, optionsOrPredicate?: Function|PageWaitForEventOptionsOrPredicate): Promise<Object>;
```

**Comment by pavelfeldman** [comment]
I thought we already had it. I can see specializations in my working folder `types/types.d.ts`, but not in the playwright-core@next. @JoelEinbinder do you know what is going on here?

**Comment by JoelEinbinder** [comment]
This was fixed in the latest release! Please try it out, and comment here or open a new issue if there are any problems with it.

**Comment by JoelEinbinder** [comment]
Fixed via #1166

**Comment by JoelEinbinder** [comment]
I misspoke, the PR seems to have gotten lost in the mail. Thanks for catching this!
